spec/features/participant-declarations/submit_participant_declarations_spec.rb

```diff
@@ -0,0 +1,178 @@
+    delivery_partner = create(:delivery_partner)
+    create(:partnership,
+           school: ect_profile.school,
+           lead_provider: @cpd_lead_provider.lead_provider,
+           cohort: ect_profile.cohort,
+           delivery_partner: delivery_partner)
+
+    @participant_id = ect_profile.user.id
+  end
+
+  def given_an_ecf_mentor_has_been_entered_onto_the_dfe_service
+    @mentor_id = create(:mentor_profile).user.id
+  end
+
+  def given_an_npq_participant_has_been_entered_onto_the_dfe_service
+    npq_lead_provider = create(:npq_lead_provider, cpd_lead_provider: @cpd_lead_provider)
+    npq_course = create(:npq_course, identifier: "npq-leading-teaching")
+    @npq_profile_id = create(:npq_validation_data,
+                             npq_lead_provider: npq_lead_provider,
+                             npq_course: npq_course).user.id
+  end
+
+  def when_the_participant_details_are_passed_to_the_lead_provider
+    @session.get("/api/v1/participants",
+                 headers: { "Authorization": "Bearer #{@token}" })
+    participants = JSON.parse(@session.response.body).dig("data").map { |participant| participant["id"] }
+    expect(participants.first).to eq(@participant_id)
+  end
+
+  def and_the_lead_provider_submits_a_declaration_for_the_participant_using_the_same_unique_id(participant_id, course_identifier)
+    params = common_params(participant_id, course_identifier)
+    submit_request(params)
+  end
+
+  def and_the_lead_provider_submits_a_declaration_for_the_participant_using_the_same_unique_id_with_error(course_identifier)
+    params = common_params("", course_identifier)
+    params["data"]["attributes"].reject! { |a| a["participant_id"] }
+    submit_request(params)
+  end
+
+  def then_the_declaration_made_is_valid
+    expect(ParticipantDeclaration.find(@response["id"])).to be_present
+  end
+
+  def then_the_declaration_made_is_invalid
+    expect(@response["bad_or_missing_parameters"]).not_to be_empty
+  end
+
+  def and_the_lead_provider_receives_a_response_to_confirm_that_the_declaration_was_successful
+    expect(@response_http_code).to eq(200)
+  end
+
+  def and_the_lead_provider_receives_a_response_to_confirm_that_the_declaration_has_a_validation_error
+    expect(@response_http_code).to eq(422)
+  end
+
+  # helper methods
+
+  def setup
+    lead_provider = create(:lead_provider)
+    @cpd_lead_provider = create(:cpd_lead_provider, lead_provider: lead_provider)
+    @token = LeadProviderApiToken.create_with_random_token!(cpd_lead_provider: @cpd_lead_provider)
+    @session = ActionDispatch::Integration::Session.new(Rails.application)
+
+    @params = common_params(@participant_id, "ecf-induction")
+  end
+
+private
+
+  def submit_request(params)
+    @response_http_code = @session.post("/api/v1/participant-declarations",
+                                        params: params,
+                                        headers: { "Authorization": "Bearer #{@token}" })
```

Comment:
I think we should use `body` rather than `params` - I think `params` will make url-encoded version fo what you give it, which is not the end of the world, but it is also not what we expect from LPs.

Reply:
I think it's slightly different functionality, there is no body https://www.rubydoc.info/docs/rails/4.1.7/ActionDispatch/Integration/RequestHelpers#post-instance_method